src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -1316,6 +1329,55 @@ public void WriteTo(Utf8JsonWriter writer)
             _parent.WriteElementTo(_idx, writer);
         }
 
+        /// <summary>
+        /// Write the property name of this <see cref="JsonElement"/> to a <see cref="Utf8JsonWriter"/>
+        /// in an allocation-less way, if the name is shorter than <see cref="JsonConstants.StackallocByteThreshold"/>
+        /// or it doesn't require unescaping, and the underlying buffer is long enough.
+        /// </summary>
+        /// <param name="writer">Utf8JsonWriter to write</param>
+        internal void WritePropertyNameTo(Utf8JsonWriter writer)
+        {
+            ReadOnlySpan<byte> rawName = GetRawPropertyName();
+            int firstBackSlashIndex = rawName.IndexOf(JsonConstants.BackSlash);
+
+            if (firstBackSlashIndex >= 0)
```

Comment:
Most of `JsonElement` acts as thin wrapper around `JsonDocument` implementation. Could this just be a stub calling a `JsonDocument.WritePropertyName(int index, Utf8JsonWriter writer);` method?

Reply:
> calling a `JsonDocument.WritePropertyName(int index, Utf8JsonWriter writer);`

`JsonDocument.WritePropertyName`, then `JsonDocument.UnescapeString` always rent an array for property names, regardless of how long they are or whether they need unescaping.

https://github.com/dotnet/runtime/blob/05c5144d9a3c26ac7f3576776b62e29a64a4b7a8/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs#L856C7-L865C36

Do you think we should change the logic in `JsonDocument.WritePropertyName` to make use of `stackalloc` and look for backslashes? I sincerely appreciate your thorough review.